src/caret_analyze/plot/bokeh/pub_sub_info.py

```diff
@@ -117,30 +117,46 @@ def _create_frequency_df(
         xaxis_type: str,
         pub_sub: Union[Publisher, Subscription]
     ) -> pd.DataFrame:
-        df = pub_sub.to_dataframe()
+        # The frequency should be calculated at system time, so no conversion should be done here.
+        # df = pub_sub.to_dataframe()
+        # if xaxis_type == 'sim_time':
+        #     convert_df_to_sim_time(self._get_converter(), df)
+
+        min_time, max_time = self._get_earliest_timestamp()
+        frequency = Frequency(pub_sub.to_records())
+        frequency_records = frequency.to_records(base_timestamp=min_time, until_timestamp=max_time)
+        frequency_df = frequency_records.to_dataframe()
+
+        frequency_df.rename(
+            columns={
+                frequency_df.columns[0]: self._get_ts_column_name(pub_sub),
+                frequency_df.columns[1]: 'frequency [Hz]',
+            },
+            inplace=True
+        )
+
         if xaxis_type == 'sim_time':
-            convert_df_to_sim_time(self._get_converter(), df)
+            # Convert only the column of measured time to simtime.
+            converter = self._get_converter()
+            for i in range(len(frequency_df)):
+                frequency_df.iat[i, 0] = converter.convert(frequency_df.iat[i, 0])
 
-        initial_timestamp = self._get_earliest_timestamp()
-        ts_series, freq_series = get_freq_with_timestamp(df.iloc[:, 0],
-                                                         initial_timestamp)
-        frequency_df = pd.DataFrame(data={
-            self._get_ts_column_name(pub_sub): ts_series,
-            'frequency [Hz]': freq_series
-        })
         frequency_df = add_top_level_column(frequency_df,
                                             pub_sub.topic_name)
 
         return frequency_df
 
     def _get_earliest_timestamp(
```

Comment:
This looks better for `_get_timestamp_range`.

Reply:
addressed in :[43626d3](https://github.com/tier4/CARET_analyze/pull/201/commits/43626d312470c5b83b96289ca9b0372df6fa1af0)